src/midi/fluid_midi.c

```diff
@@ -1719,8 +1816,8 @@ new_fluid_player(fluid_synth_t *synth)
     player->end_pedals_disabled = 0;
     player->last_callback_ticks = -1;
     fluid_atomic_int_set(&player->seek_ticks, -1);
-    fluid_player_set_playback_callback(player, fluid_synth_handle_midi_event, synth);
-    fluid_player_set_tick_callback(player, NULL, NULL);
+    fluid_player_set_playback_callback(player, fluid_synth_handle_midi_event, player);
```

Comment:
This effectively passes the `player` object to `fluid_synth_handle_midi_event`, which however expects a `synth` object. This will effectively crash fluidsynth, which our failing CI builds also prove. It also means that your statement to have "thoroughly tested this" is not credible!